docs/dictionary.rst

```diff
@@ -0,0 +1,1101 @@
+------------
+	A read-only file that represents a filesystem that contains some sort of code and that code's dependencies. A Docker image can be created using the `docker build` command in conjunction with a :ref:`dict Dockerfile`. If a workflow language references a Docker image, then the workflow executor will download that Docker image (unless was already downloaded previously) and add a writable layer onto the Docker image, which results in the creation of a :ref:`dict Docker container`.  
+
+
+.. updated 2022-05-16  
+
+
+
+.. _dict Dockerfile:
+
+Dockerfile
+----------
+	A file describing the creation of a :ref:`dict Docker image` by running commands that each form a :ref:`dict layer`.  
+
+Further reading: `<https://docs.docker.com/engine/reference/builder/>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict Dockstore CLI:
+
+Dockstore CLI
+-------------
+*abbreviation for* Dockstore Command Line Interface  
+
+	A command-line program developed by Dockstore. It is not required to use Dockstore, but it has many features to make running and developing workflows easier.  
+
+see also :ref:`dict CLI`  
+
+Further reading: `<https://docs.dockstore.org/en/stable/advanced-topics/dockstore-cli/dockstore-cli-faq.html>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict Dockstore GitHub App:
+
+Dockstore GitHub App
+--------------------
+	The GitHub App that allows for Dockstore to communicate with GitHub repositories.  
+
+see also :ref:`dict GitHub App registration`  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict DOI:
+
+DOI
+---
+*abbreviation for* Digital Object Identifier  
+
+	An identifier that provides a long-lasting link to some sort of :ref:`dict immutable` digital object. On Dockstore, you can use Zenodo to mint a DOI of your workflows and tools to increase reproducibility.  
+
+
+.. updated 2022-05-16  
+
+
+
+.. _dict DS-I Africa:
+
+DS-I Africa
+-----------
+*abbreviation for* Data Science for health discovery and Innovation in Africa  
+
+	An :ref:`dict NIH` initiative to leverage data science to address the African continent's public health needs.  
+
+Further reading: `<https://commonfund.nih.gov/africadata>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict EC2:
+
+EC2
+---
+*abbreviation for* Elastic Compute Cloud  
+
+	The cloud computing side of :ref:`dict AWS`. When running workflows on these backends, disk size will scale with your workflow requirements automatically. EC2 instances allow you to make use of Amazon's :ref:`dict spot instance` feature, which may reduce the cost of running workflows.  
+
+Further reading: `<https://docs.aws.amazon.com/ec2/index.html>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict egress:
+
+egress
+------
+pronounced "ee-gress", rhymes with aggress  
+
+	The action of leaving a place. In the context of :ref:`dict cloud computing`, an egress charge is a fee charged for downloading a file. Sometimes, the person hosting the file is charged for data egress. Other times, the person downloading the file is charged.  
```

Comment:
Since "egress" shows up in so many forms (data egress, egress charge, egress fee, on its own such as "we have to pay a lot for egress"), I'm leaning towards keeping the definition's title as egress. That being said your suggested definition is more technically accurate/overall helpful than what I wrote. How about this?

> The action of leaving a place. In the context of [cloud computing], data egress refers to data being moved from one location to another, such as from the cloud to a local machine, between cloud providers, and between locations of a single cloud provider. Data egress often results in the charge of fees (usually called egress charges). Data egress can be one of the most expensive cloud costs incurred. Sometimes, the person hosting the file is charged for data egress. Other times, the person downloading the file is charged (such as when downloading files from a Google bucket that has the requester-pays option enabled).

Reply:
LGTM